src/main/java/teammates/ui/constants/ApiStringConst.java

```diff
@@ -0,0 +1,40 @@
+package teammates.ui.constants;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import teammates.common.util.FieldValidator;
+
+/**
+ * Special constants used by the back-end.
+ */
+public enum ApiStringConst {
+    // CHECKSTYLE.OFF:JavadocVariable
+    EMAIL_REGEX(escapeRegex(FieldValidator.REGEX_EMAIL));
```

Comment:
If combined, the generated `ApiConst` enum in `./api-const.ts` would be a heterogeneous enum. I was concerned that this might introduce ambiguity and potential for bugs, so I put the string and number constants into separate enums. However, if it is OK for `ApiConst` to contain both `string` and `number`, I can put the email regex together with `ApiConst`.

----

**Example 1 (bug)**: [notification-edit-form.component.ts line 25](https://github.com/TEAMMATES/teammates/blob/master/src/web/app/pages-admin/admin-notifications-page/notification-edit-form/notification-edit-form.component.ts#L25)

```ts
  NOTIFICATION_TITLE_MAX_LENGTH = ApiConst.NOTIFICATION_TITLE_MAX_LENGTH;
```

Leaving this line as is results in a compilation error as the type inferred for `NOTIFICATION_TITLE_MAX_LENGTH` is not specifically `number`:
<img width="1295" alt="screenshot of compile error" src="https://github.com/TEAMMATES/teammates/assets/40383042/0c72f083-b0cb-4fc5-91cb-f92c3acf0ecd">

It can be fixed by explicitly specifying the type `number` when declaring the variable, or changing it to `readonly`:
```ts
  NOTIFICATION_TITLE_MAX_LENGTH: number = ApiConst.NOTIFICATION_TITLE_MAX_LENGTH;
// or
  readonly NOTIFICATION_TITLE_MAX_LENGTH: number = ApiConst.NOTIFICATION_TITLE_MAX_LENGTH;
```

This seems to be an outlier, as the other components that similarly declare the API constants do use `: number` and/or `readonly` and work as expected.

**Example 2 (working)**: [instructor-search-page.component.ts line 69](https://github.com/TEAMMATES/teammates/blob/master/src/web/app/pages-instructor/instructor-search-page/instructor-search-page.component.ts#L69). This code uses the constants directly (`ApiConst.XYZ`) and works properly as the type is inferred correctly.
```ts
          if (searchStudentsTable.length >= ApiConst.SEARCH_QUERY_SIZE_LIMIT) {
```

Reply:
I see, this seems reasonable.